comp/otelcol/ddflareextension/impl/configstore_test.go

```diff
@@ -0,0 +1,199 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package ddflareextensionimpl defines the OpenTelemetry Extension implementation.
+package ddflareextensionimpl
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	collectorcontribimpl "github.com/DataDog/datadog-agent/comp/otelcol/collector-contrib/impl"
+	converterimpl "github.com/DataDog/datadog-agent/comp/otelcol/converter/impl"
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/datadogexporter"
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/processor/infraattributesprocessor"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/datadogconnector"
+	"go.opentelemetry.io/collector/component/componenttest"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/confmaptest"
+	"go.opentelemetry.io/collector/confmap/converter/expandconverter"
+	"go.opentelemetry.io/collector/confmap/provider/envprovider"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+	"go.opentelemetry.io/collector/confmap/provider/httpprovider"
+	"go.opentelemetry.io/collector/confmap/provider/httpsprovider"
+	"go.opentelemetry.io/collector/confmap/provider/yamlprovider"
+	"go.opentelemetry.io/collector/otelcol"
+	"gopkg.in/yaml.v2"
+)
+
+// this is only used for config unmarshalling.
+func addFactories(factories otelcol.Factories) {
+	factories.Exporters[datadogexporter.Type] = datadogexporter.NewFactory(nil, nil, nil, nil, nil)
+	factories.Processors[infraattributesprocessor.Type] = infraattributesprocessor.NewFactory(nil, nil)
+	factories.Connectors[component.MustNewType("datadog")] = datadogconnector.NewFactory()
+	factories.Extensions[Type] = NewFactory(nil, otelcol.ConfigProviderSettings{})
+}
+
+func TestGetConfDump(t *testing.T) {
+	// get factories
+	cc := collectorcontribimpl.NewComponent()
+	factories, err := cc.OTelComponentFactories()
```

Comment:
May be this might work (as there no indirects) please for BYOC workflow by adding a component . https://dd.slack.com/archives/C06S6HQSK3Q/p1724785499015999

Reply:
I was already statically defining factories for another test, so used that instead.